src/tools/evaluate.ts

```
@@ -42,26 +41,56 @@ const evaluate = defineTabTool({
-      // whose `toString()` returns the user-supplied source instead.
-      const func = new Function() as any;
-      func.toString = () => params.function;
-      const result = locator ? await locator.evaluate(func) : await tab.page.evaluate(func);
-      response.addResult(JSON.stringify(result, null, 2) || 'undefined');
-    });
+    try {
+      await tab.waitForCompletion(async () => {
+        let evalResult: { result: unknown, isFunction: boolean };
+        if (locator) {
+          evalResult = await locator.evaluate(async (element, source) => {
+            const value = eval(`(${source})`);
```

Biome (2.5.1)

[error] eval() exposes to security risks and performance issues.

(lint/security/noGlobalEval)